ipyparallel/cluster/shellcmd.py

```diff
@@ -0,0 +1,859 @@
+#!/usr/bin/env python
+"""
+Shell helper application for OS independent shell commands
+
+Currently, the following command are supported:
+* start:   starts a process into background and returns its pid
+* running: check if a process with a given pid is running
+* kill:    kill a process with a given pid
+* mkdir:   creates directory recursively (no error if directory already exists)
+* rmdir:   remove directory recursively
+* exists:  checks if a file/directory exists
+* remove:  removes a file
+"""
+
+import base64
+import enum
+import inspect
+import os
+import re
+import shlex
+import sys
+import time
+from datetime import datetime
+from random import randint
+from subprocess import CalledProcessError, Popen, TimeoutExpired, check_output
+from tempfile import NamedTemporaryFile
+
+
+class Platform(enum.Enum):
+    Unknown = 0
+    Linux = 1
+    Windows = 2
+    MacOS = 3
+
+    @staticmethod
+    def get():
+        import sys
+
+        tmp = sys.platform.lower()
+        if tmp == "win32":
+            return Platform.Windows
+        elif tmp == "linux":
+            return Platform.Linux
+        elif tmp == "darwin":
+            return Platform.MacOS
+        else:
```

Comment:
your are right - didn't work on freebsd.
A classification in Windows and Posix would definitely work. Personally, MacOS feels so much differently to me which is why I would keep the MacOS enum. Nevertheless, it seems that MacOS is very close to an 
[posix system](https://stackoverflow.com/questions/5785516/is-mac-os-x-a-posix-os) so final I leave the final decision up to you. _Dual_ enum?

Reply:
macOS is a BSD-based posix system. Since this enum is here for your own decisions, there's no reason to keep it unless you have any `if macos:` branches. If there aren't any, it's not a useful distinction _in this context_.